File: BOJ(Java 공부용)/15651.java
import java.io.*;
import java.util.ArrayList;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);
        ArrayList<Integer> v = new ArrayList<>();
        go(0,n,m,v);
        bw.flush();
        bw.close();
    }
    private static void go(int cnt,int n,int depth,ArrayList<Integer> v) throws IOException {
        if(cnt == depth) {
            for(int i=0;i<v.size();i++) {
                bw.write(v.get(i)+" ");
            }
            bw.write("\n");
            return;
        }
        for(int i=0;i<n;i++) {
                v.add(i+1);
                go(cnt+1,n,depth,v);
                v.remove(v.size()-1);
        }
    }
}

File: BOJ(Kotlin)/15651.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

val bw = BufferedWriter(OutputStreamWriter(System.out))
fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()
    val array = mutableListOf<Int>()
    dfs(n,m,0,array)
    bw.flush()
    bw.close()
}

fun dfs(n : Int, m : Int,count : Int, array : MutableList<Int>)  {
    if (count == m) {
        for (value in array) {
            bw.write("$value ")
        }
        bw.write("\n")
        return
    }

    for (i in 0 until n) {
            array.add(i+1)
            dfs(n,m,count+1,array)
            array.removeLast()
    }

}